caproto/tests/test_record_compliance.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+# we are going to test the mock 'ai' record from ioc_examples/mocking_records
+# and see if:
+# - the types match the "real" 'ai' record
+# - all the .* PVs can be read
+# - all the enums of the .* can be read
+import time
+
+import epics
+from epics import ca
+import pytest
+
+from . import conftest
+
+
+# type equality between pyepics and "real" epics, **this is assumed **
+type_equal = {
+    "DBF_STRING": ["string"],
+    "DBF_CHAR": ["char"],
+    "DBF_UCHAR": ["char"],
+    "DBF_SHORT": ["short"],
+    "DBF_USHORT": ["short"],
+    "DBF_LONG": ["int", "long"],
+    "DBF_ULONG": ["int", "long"],
+    "DBF_INT64": ["long"],
+    "DBF_UINT64": ["long"],
+    "DBF_FLOAT": ["float"],
+    "DBF_DOUBLE": ["double"],
+    "DBF_ENUM": ["enum"],  # have an associated set of ASCII strings
+    "DBF_MENU": ["enum"],  # have an associated set of ASCII strings
+    "DBF_DEVICE": ["enum"],  # have an associated set of ASCII strings
+    "DBF_INLINK": ["string"],  # is a link structure
+    "DBF_OUTLINK": ["string"],  # is a link structure
+    "DBF_FWDLINK": ["string"],  # is a link structure
+    "DBF_NOACCESS": [""]  # for private use by record processing routines
+}
+
+
+def test_record_compliance(request, prefix, record_type_name,
+                           record_type_to_fields):
+    fields = record_type_to_fields[record_type_name]
+
+    # host a caproto record to test(you could also try this test on a real record).
+    # NOTE: using the below will actually run a seperate program from command line,
```

Comment:
The fixture uses `sys.executable` so it should be running in the same vevn as the tests are running in.

Reply:
Definitely. I didn't update these comments from the original PR.